finesse/FINESSE_USAGE.md

```diff
@@ -0,0 +1,82 @@
+# Design of the Finesse Benchmark Tool
+
+This tool simplifies the process of comparing different search engines and
+assessing their accuracy. It's designed to be straightforward, making it easy
+to understand and use.
+
+## How it Works
+
+- **Single command:**
+  - Users can enter commands with clear instructions to choose a search engine,
+    specify a directory for JSON files and specify the backend URL.
+  - Mandatory arguments:
+    - `--engine [search engine]`: Pick a search engine.
+      - `ai-lab` : AI-Lab search engine
+      - `azure`: Azure search engine
+      - `static`: Static search engine
+      - `llamaindex`: LlamaIndex search engine
+    - `--path [directory path]`: Point to the directory with files structured
+    - `--host [API URL]`: Point to the finesse-backend URL
+      with JSON files with the following properties:
+      - `score`: The score of the page.
+      - `crawl_id`: The unique identifier associated with the crawl table.
+      - `chunk_id`: The unique identifier of the chunk.
+      - `title`: The title of the page.
+      - `url`: The URL of the page.
+      - `text_content`: The main textual content of the item.
+      - `question`: The question to ask.
+      - `answer`: The response to the asked question.
+  - Optional argument:
+    - `--format [file type]`:
+      - `csv`: Generate a CSV document
+      - `md`: Generate a Markdown document, selected by default
+    - `--once`: Go through all the json files and does not repeat
+    - `--top`: Limit the number of results returned by the search engine
+- **Many tests**
+  - Test all the JSON files in the path directory
+- **Accuracy score**
+  - The tool compares the expected page with the actual Finesse response pages.
+  - Calculates an accuracy score for each response based on its position in the
+    list of pages relative to the total number of pages in the list. 100% would
+    correspond of being at the top of the list, and 0% would mean not in the
+    list.
+- **Round trip time**
+  - Measure round trip time of each request
+- **Summary statistical value**
+  - Measure the average, median, standard deviation, minimum and maximal accuracy scores and round trip time
+
+## Diagram
+
+![Alt text](diagram.png)
```

Comment:
where's the source of this diagram? Mermaid? need the source version-controlled.

Reply:
Removed the picture and added the source of the diagram